tests/DataTransferObjectTest.php

```diff
@@ -421,11 +421,38 @@ public function iterable_is_supported()
         $this->markTestSucceeded();
     }
 
+    /** @test */
+    public function key_value_arrays_value_is_type_checked()
+    {
+        new class(['map' => ['a' => 1, 'b' => 2]]) extends DataTransferObject {
+            /** @var array<string, int> */
+            public $map;
+        };
+
+        $this->expectException(DataTransferObjectError::class);
+
+        new class(['map' => ['a' => 1, 'b' => 'wrong']]) extends DataTransferObject {
+            /** @var array<string, int> */
+            public $map;
+        };
+    }
+
+    /** @test */
+    public function key_value_arrays_key_is_type_checked()
+    {
+        $this->expectException(DataTransferObjectError::class);
+
+        new class(['map' => [1 => 1, 'b' => 2]]) extends DataTransferObject {
+            /** @var array<string, int> */
```

Comment:
Maybe also add a test showing the `array<string,int>` (i.e _no_ space) also works. Just in case.